src/services/profileService.ts

```diff
@@ -64,6 +64,7 @@ export default class ProfileService {
     upsertProfile: ExtendedProfile
   ): Promise<ProfileWithEmail> {
     return new Promise(async (resolve, reject) => {
+      // tslint:disable-next-line:no-use-of-empty-return-value await-promise
```

Comment:
I'm not sure why this rule is triggered, does `upsertProfile` return a value?

Reply:
yes upsertProfile returns a Promise in the same way as `getProfile` at line 32 in the same file, but sonar mark only this as error...

Reply:
I think there is a problem here with the fact that you're passing an async function ti the `Promise` constructor (AFAIK you can't)